# 📊 Parcial – Análisis de Redes Sociales (Primer Corte)

## 🎯 Objetivo
Evaluar tu dominio de **filtros** y **agrupaciones** en Python (sin pivots ni gráficos) y tu capacidad de **análisis de negocio** en métricas de marketing: **CAC, LTV, churn rate, funnel, unit economics (LTV/CAC)**.

---
## 🧩 Contexto de Negocio (Caso Real)
**StartUp SaaS en crecimiento.** Vende planes **Basic, Pro, Enterprise** en **LatAm, North America, Europe, APAC**. Los canales de adquisición son: **meta_ads, google_ads, organic_search, outbound_sales**. Los usuarios se registran por **web** o **mobile_app**.

La dirección quiere decidir **dónde invertir el presupuesto del próximo trimestre**. Te piden:

1) Identificar **canales más rentables** (relación **LTV/CAC** y churn).  
2) Detectar **regiones** con mayor **potencial de crecimiento** y/o **riesgo**.  
3) Evaluar el desempeño por **tier** (Basic, Pro, Enterprise).  
4) Recomendar **asignación de presupuesto** por **canal + tier**.

---
## 📂 Instrucciones
- Trabaja únicamente con **filtros** y **agrupaciones**. **No uses** tablas dinámicas (pivot), merges ni gráficos.
- Archivo a usar: `clientes_marketing.csv`.
- Escribe **código + interpretación breve** para cada inciso.
- Si el enunciado dice **“Agrupa”**, usa `groupby`. Si dice **“Filtra”**, usa filtrado con máscaras booleanas. Si dice **“Ambas”**, realiza primero el **filtro** y luego la **agrupación**.

---


## Descargar Tabla (Correr una vez)

In [4]:
!wget https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/cac_ltv_model.csv

--2025-09-17 23:29:21--  https://github.com/javierherrera1996/IntroMarketingAnalytics/raw/refs/heads/main/PrimerCorte/cac_ltv_model.csv
Resolving github.com (github.com)... 140.82.116.3
Connecting to github.com (github.com)|140.82.116.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/cac_ltv_model.csv [following]
--2025-09-17 23:29:22--  https://raw.githubusercontent.com/javierherrera1996/IntroMarketingAnalytics/refs/heads/main/PrimerCorte/cac_ltv_model.csv
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 661398 (646K) [text/plain]
Saving to: ‘cac_ltv_model.csv’

cac_ltv_model.csv   100%[===================>] 645.90K  --.-KB/s

## Importar Datos

In [5]:
import pandas as pd
pd.set_option('display.float_format', lambda x: f'{x:,.2f}')

# 1) Cargar dataset
data = pd.read_csv('cac_ltv_model.csv')


### ✅ Variables relevantes (recordatorio)
- `acquisition_channel`, `signup_source`, `region`, `customer_tier`
- `plan_price`, `discount_rate`, `arpu`, `gross_margin`, `churn_rate`
- `contract_length_months`, `marketing_spend`

### 🧮 Fórmulas de negocio (a usar en el examen)
- **CAC por canal**:  
    $$ CAC = \frac{\text{Total Marketing Spend}}{\text{Clientes adquiridos}} $$
  
- **LTV por observación** (aprox.):  
   $$ LTV = ARPU \times contract\_length\_months \times gross\_margin \times (1 - churn\_rate) $$

- **Unit economics**:  
  $$ LTV/CAC   $$(ideal \> 3 como regla general de SaaS)

---



### Parte Exploratoria: Cree una muestra de los primeros 5 elementos de esta tabla. ¿Cuantas observaciones tienes?

In [6]:
data.head(5)

year  month    date  customer_id acquisition_channel signup_source  \
0  2023      1  Jan-23         1001      outbound_sales           web   
1  2023      1  Jan-23         1002            meta_ads           web   
2  2023      1  Jan-23         1003      organic_search    mobile_app   
3  2023      1  Jan-23         1004      organic_search           web   
4  2023      1  Jan-23         1005      organic_search           web   

          region customer_tier  plan_price  discount_rate   arpu  \
0  North America         Basic       78.84           0.10  63.63   
1          LatAm           Pro      120.23           0.00 124.47   
2  North America    Enterprise      335.82           0.20 278.20   
3         Europe           Pro      193.89           0.10 190.65   
4           APAC    Enterprise      471.80           0.10 445.32   

   gross_margin  churn_rate  contract_length_months  marketing_spend  
0          0.76        0.02                      12           212.48  
1          0.84        0.05                       1           352.60  
2          0.81        0.02                      12            55.40  
3          0.80        0.08                       1            49.10  
4          0.80        0.08                       1            45.27

## 🔎 Parte A – Filtros

hint: Crea una tabla con el filtro y luego agrupa la variable que te piden.

1. **Filtra** los clientes del canal **`meta_ads`**. Calcula el **ARPU promedio** de este subconjunto.  



In [19]:
data[data["acquisition_channel"]== "meta_ads"]["arpu"].mean()

np.float64(168.78858272162617)

2. **Filtra** clientes de **LatAm** con **`churn_rate > 0.05`**. ¿Cuántos son? ¿Cuál es su **ARPU promedio**?  

In [172]:
data[(data["region"] == "LatAm") & (data["churn_rate"] > 0.05)] ["arpu"].mean()

np.float64(174.78453658536586)

3. **Filtra** clientes **Enterprise** con `contract_length_months > 6`. ¿Cuál es su **gross_margin promedio**?

In [164]:
data[(data["customer_tier"] == "Enterprise") & (data["contract_length_months"] > 6)]["gross_margin"].mean()

np.float64(0.8248397435897435)

## 📊 Parte B – Agrupaciones (SOLO agrupar)
4. **Agrupa** por `acquisition_channel` y calcula **ARPU promedio**. Ordena de mayor a menor.  


In [157]:
data.groupby("acquisition_channel")["arpu"].mean().sort_values()

acquisition_channel
meta_ads         168.79
organic_search   171.14
google_ads       172.37
outbound_sales   173.61
Name: arpu, dtype: float64

de mayor a menor seria asi

outbound_sales	173.61

google_ads	172.37

organic_search	171.14

meta_ads	168.79

5. **Agrupa** por `region` y calcula **churn_rate promedio**. Identifica la región con mayor churn.  



In [37]:
data.groupby("region")["churn_rate"].mean()

region
APAC            0.05
Africa          0.05
Europe          0.05
LatAm           0.05
Middle East     0.05
North America   0.05
Name: churn_rate, dtype: float64

Ninguna

6. **Agrupa** por `customer_tier` y calcula **marketing_spend total**. ¿Cuál tier consume más presupuesto?

In [39]:
data.groupby("customer_tier")["marketing_spend"] .sum()

customer_tier
Basic        584,419.11
Enterprise   584,348.87
Pro          589,265.04
Name: marketing_spend, dtype: float64

El tier que consume mas presupuesto es el pro con	**589,265.0**

## 🔀 Parte C – Filtro **y** Agrupación (Ambas)
7. **Filtra** solo registros de **`signup_source = 'web'`** y luego **agrupa** por `acquisition_channel` para obtener el **churn_rate promedio**.  


In [55]:
data[data["signup_source"]== "web"] .groupby("acquisition_channel")["churn_rate"].mean()

acquisition_channel
google_ads       0.05
meta_ads         0.05
organic_search   0.05
outbound_sales   0.05
Name: churn_rate, dtype: float64

8. **Filtra** solo **`mobile_app`** y luego **agrupa** por `region` para calcular **ARPU promedio**.  


In [81]:
data[data["signup_source"]== "mobile_app"] .groupby("region")["arpu"].mean()

region
APAC            169.09
Africa          172.60
Europe          165.89
LatAm           173.20
Middle East     171.71
North America   174.00
Name: arpu, dtype: float64

9. **Filtra** a clientes **`Pro`** y **agrupa** por `acquisition_channel` para calcular **marketing_spend total**.


In [61]:
data[data["customer_tier"]== "pro"] .groupby("acquisition_channel")["marketing_spend"].sum()

Series([], Name: marketing_spend, dtype: float64)

## 📈 Parte D – Métricas de negocio (CAC, LTV, LTV/CAC)
10. **CAC por canal (Agrupa)**: calcula el CAC de cada `acquisition_channel` como:  
   `CAC = marketing_spend_total_del_canal / #clientes_del_canal`  


In [125]:
a= data.groupby("acquisition_channel")["marketing_spend"].sum()

In [126]:
b= data.groupby("acquisition_channel")["customer_id"].count()

In [174]:
pepita= a/b

11. **CAC por tier (Agrupa)**: calcula el CAC de cada `customer_tier` como:  
   `CAC = marketing_spend_total_del_tier / #clientes_del_canal`  


In [131]:
c= data.groupby("customer_tier")["marketing_spend"].sum()

In [130]:
d= data.groupby("customer_tier")["customer_id"].count()

In [158]:
c/d

customer_tier
Basic        249.75
Enterprise   249.19
Pro          248.43
dtype: float64

12. **LTV por tier (Ambas)**: crea una columna `ltv_individual` con la fórmula dada.


In [141]:
data ["ltv_individual"]= data["arpu"] * data["contract_length_months"] * data["gross_margin"] * (1 - data['churn_rate'])

12. A. Luego **agrupa** `ltv_individual` por
`acquisition_channel` para obtener el **LTV promedio**.  

In [160]:
data.groupby("acquisition_channel")["ltv_individual"].mean()

acquisition_channel
google_ads       710.25
meta_ads         729.05
organic_search   765.13
outbound_sales   751.57
Name: ltv_individual, dtype: float64

12. B. Luego **agrupa** `ltv_individual`por `customer_tier` para obtener el **LTV promedio**.





In [162]:
lifetimevalue= data.groupby("customer_tier")["ltv_individual"].mean()

13. **Unit economics (Ambas)**: combina tus resultados para comparar **LTV promedio por tier** contra **CAC por canal** y comenta **qué combinaciones canal + tier** lucen más saludables (busca **LTV/CAC > 3**).


In [163]:
lifetimevalue/pepita

Basic            NaN
Enterprise       NaN
Pro              NaN
google_ads       NaN
meta_ads         NaN
organic_search   NaN
outbound_sales   NaN
dtype: float64

## 🧠 Parte E – Análisis crítico (respuesta abierta)
13. **CMO por un día**: con tus resultados, ¿en qué **canal(es)** invertirías más el próximo trimestre? ¿Por qué? (Cita **CAC** y **churn**).  
14. ¿Qué **región** ves con mayor riesgo? ¿Qué hipótesis explicarían su **churn**?  
15. Identifica **dos combinaciones canal + tier** con mejor **LTV/CAC** y propón **dos acciones** concretas de optimización (p. ej., creatividades, audiencias, pricing, retención).

---
✍️ **Nota**: Mantente disciplinado con el enunciado (**Filtra**, **Agrupa**, **Ambas**). La evaluación pondera **correctitud técnica** y **calidad de interpretación de negocio**.


El canal que menos me cuesta es organic- serach, me cuesta menos adiquir clientes por este canal a difrencia de los otros, invertiria mas en este porque es el mas eficiente, el CAC es de 171.14 y el churn es de 0.05

la region que tiene menos ingresos por usuario (arpu) seria APAC	169.09 que es el numero mas bajito de ventas, APAC estaria en riesgo, si lo medimos por churn Ninguna region esta en gran riesgo, todas tienen un church de 00.5, son pocos los clientes que se van en un tiempo especifico, quizas los clientes son muy fieles a la compañia, aunque no generen tantas ganancias

1. organic- search con pro

-  se pueden Crear más contenidos en Google que atraigan a usuarios interesados en el plan Pro

-  Acompañar mejor a los nuevos clientes (guias, recordatorios, ayuda inicial) para que se queden más tiempo

2. outbound - sales con enterprise

- Ofrecer descuentos si compran planes por un año o más

- Enfocar las llamadas y esfuerzos de ventas en empresas grandes que generen más ingresos